sdk/analytics/src/test/java/com/klaviyo/analytics/networking/KlaviyoApiClientTest.kt

```diff
@@ -507,6 +508,60 @@ internal class KlaviyoApiClientTest : BaseTest() {
         verify(exactly = 1) { request2.send(any()) }
     }
 
+    @Test
+    fun `503 results in a retry using exponential backoff`() {
+        // same as above test but with a 503 instead of a 429
+        // First unsent request, which we will retry till max attempts (calculated by exponential)
+        // note that we are not sending the retry-after header on a 503
+        val request1 = mockRequest("uuid-retry", KlaviyoApiRequest.Status.Unsent, 503)
+
+        every { request1.state } answers {
+            when (request1.attempts) {
+                0 -> KlaviyoApiRequest.Status.Unsent.name
+                20 -> KlaviyoApiRequest.Status.Failed.name
```

Comment:
Where does 20 come from?

Reply:
Was just testing some arbitrary values, figured this wasn't explicitly testing the limits so having it attempt less than 50 was fine. If you want I can revert this tho

Reply:
Ooh sorry i read this wrong, this is the mocked response. Yeah this is fine!